### PR TITLE
fix(models): pass quant_config to eh_proj in MTP layers to prevent silent precision loss

### LINKED ISSUE #48492: [Bug]: Missing quant_config in MTP eh_proj causes severe silent precision loss during W8A8 quantization
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>

### 🐛 Describe the bug

When running models with Multi-Token Prediction (MTP) enabled under W8A8 (or similar) quantization, severe silent precision loss occurs. 
Through dump analysis comparing original (BF16) and W8A8 weights, we discovered that the `eh_proj` linear layer in multiple MTP implementations uses standard `nn.Linear` without passing the `quant_config`. This causes the quantized checkpoint weights to be incorrectly loaded and computed, resulting in massive numerical distortion and degraded generation quality.
### Root Cause Analysis
In several MTP model files, `eh_proj` is defined as a standard `nn.Linear`:
```python
# BUG: nn.Linear doesn't accept quant_config
self.eh_proj = nn.Linear(config.hidden_size * 2, config.hidden_size, bias=False)
```
When W8A8 quantization is enabled:
1. The checkpoint provides INT8 quantized weights (value range `[-127, 127]`) and corresponding per-channel scales for this layer.
2. Because `nn.Linear` lacks quantization awareness (no `quant_config`), the default weight loader directly loads the INT8 integers into a BF16 tensor without applying the dequantization scales.
3. During the forward pass, `F.linear` executes standard floating-point multiplication using these massive integer values, causing the output norm to amplify by ~1116x compared to the unquantized baseline.
**Why is this a "silent" precision loss?**
The immediate next layer after `eh_proj` is `input_layernorm` (RMSNorm). Mathematically, RMSNorm normalizes the macroscopic magnitude, which cancels out the ~1116x global amplification. This prevents an immediate crash (like NaNs), making the issue hard to spot casually. 
However, the precision is permanently destroyed because:
* **Loss of per-channel proportions:** W8A8 uses per-channel quantization. Bypassing the scale destroys the relative size relationships between feature channels. RMSNorm only fixes the global norm, not the internal channel distortions.
* **BF16 precision truncation:** The massive amplification causes large numbers to collide with BF16's precision limits ("large numbers eat small numbers"), permanently discarding subtle numerical differences before normalization.
### Proposed Fix
Replace `nn.Linear` with `ReplicatedLinear` and pass the `quant_config` so the weights are correctly dequantized during forward passes. `ReplicatedLinear` is appropriate here because the input to `eh_proj` is already TP-sharded from `cat(enorm, hnorm)`, and its output goes directly into the TP-aware `mtp_block`, meaning no cross-rank communication is needed.
```diff
- self.eh_proj = nn.Linear(config.hidden_size * 2, config.hidden_size, bias=False)
+ from vllm.model_executor.layers.linear import ReplicatedLinear
+ self.eh_proj = ReplicatedLinear(
+     config.hidden_size * 2,
+     config.hidden_size,
+     bias=False,
+     quant_config=quant_config,
+     params_dtype=config.dtype,
+ )
```
### Affected Files
This bug exists in at least 7 MTP model implementations:
- `vllm/model_executor/models/hy_v3_mtp.py` (~Line 100)
- `vllm/model_executor/models/deepseek_mtp.py` (~Line 73)
- `vllm/model_executor/models/glm4_moe_lite_mtp.py` (~Line 93)
- `vllm/model_executor/models/glm4_moe_mtp.py` (~Line 89)
- `vllm/model_executor/models/glm_ocr_mtp.py` (~Line 66)
- `vllm/model_executor/models/openpangu_mtp.py` (~Line 61)
- `vllm/model_executor/models/step3p5_mtp.py` (~Line 54)
*(Note: Implementations like `longcat_flash_mtp.py` and `mimo_v2_mtp.py` already correctly use `ReplicatedLinear` with `quant_config` and serve as good references.)*
### Additional context
Applying this fix ensures that MTP layers correctly handle quantized weights, restoring the intended precision and preventing subtle output quality degradation during W8A8 inference.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by ErenAta16**
## Summary

Fixes #48492. Seven MTP (Multi-Token Prediction) model implementations define `eh_proj` as a plain `nn.Linear`, which doesn't accept `quant_config`. Under W8A8 (or similar) quantization, the default weight loader loads raw INT8 checkpoint values into a BF16 tensor without applying the dequantization scale, the following RMSNorm masks the resulting global magnitude amplification, so there's no crash, but per-channel proportions and BF16 precision are permanently destroyed before normalization, silently degrading generation quality. Full root-cause analysis is in #48492, including a dump-based comparison against the BF16 baseline.

`longcat_flash_mtp.py` and `mimo_v2_mtp.py` already use the correct pattern (`ReplicatedLinear` with `quant_config` and `prefix="eh_proj"`), this PR brings the same pattern to the seven files that didn't:

- `hy_v3_mtp.py`
- `deepseek_mtp.py`
- `glm4_moe_lite_mtp.py`
- `glm4_moe_mtp.py`
- `glm_ocr_mtp.py`
- `openpangu_mtp.py`
- `step3p5_mtp.py`

## One correction versus the fix proposed in #48492

`ReplicatedLinear` returns a `(output, bias)` tuple, unlike `nn.Linear` which returns just the tensor. The diff sketched in the issue only showed changing the layer definition; applying just that would leave every `hidden_states = self.eh_proj(...)` call site trying to use a tuple as a tensor. This PR updates the layer definition **and** the call site in each file (`hidden_states, _ = self.eh_proj(...)`) to match the working pattern in `longcat_flash_mtp.py`.

`openpangu_mtp.py` is a partial exception: `OpenPanguMultiTokenPredictorLayer` subclasses `DeepSeekMultiTokenPredictorLayer` and inherits its `forward()` rather than overriding it, so only the `__init__` needed the layer-definition change there, the inherited (now-fixed) forward from `deepseek_mtp.py` already unpacks correctly.

## Verification

- Confirmed `quant_config` is in scope at each `eh_proj` definition site in all seven files (either as a constructor parameter or a local pulled from `vllm_config.quant_config`) before making the change.
- Checked `deepseek_mtp.py`'s `_rewrite_spec_layer_name`, which does substring matching (`"eh_proj" in name`) against weight names for spec-layer remapping. `ReplicatedLinear`'s `prefix="eh_proj"` preserves this, so no change needed there.
- Grepped the diff for any remaining `nn.Linear` eh_proj definitions or un-updated call sites across all seven files, none left.

Not able to run an end-to-end W8A8 generation comparison in this environment (no GPU), so this is verified by static tracing against the known-correct reference implementations rather than a fresh empirical repro. Flagging that explicitly rather than claiming more than I checked.

### CODE REVIEW

**Comment by ErenAta16** [comment]
Noting for anyone checking why `pre-run-check` is red: it's the standard trust gate (`ready`/`verified` label or 4+ merged PRs from the author), not a code issue, pre-commit itself never ran (skipped once the gate failed). Happy to have this labeled for a full run whenever a maintainer has a chance to look at the actual diff.

**Comment by ErenAta16** [comment]
Rebased onto `fa2a2589b`; `mergeable` is back to true and it's a single commit again.

`step3p5_mtp.py` was the only conflict — its `SharedHead` gained `prefix=f"{prefix}.shared_head"` upstream in the meantime, so I kept upstream's version and only replaced `eh_proj`.

One change while I was in there. The prefix is now fully qualified — `f"{prefix}.eh_proj"` instead of the bare `"eh_proj"` the first revision used. `is_layer_skipped` matches the ignore list by exact string:

```python
>>> is_layer_skipped("eh_proj", ["model.layers.61.eh_proj"])
False
>>> is_layer_skipped("model.layers.61.eh_proj", ["model.layers.61.eh_proj"])
True
```

so the bare name can't match a `modules_to_not_convert` entry, and a layer the checkpoint asks to leave alone would get quantized instead. That's the opposite of what this PR is for, hence the change.

`longcat_flash_mtp.py:46` passes the bare `"eh_proj"` today — that's where I originally took the pattern from. Left it alone as out of scope, but flagging it since it looks like the same gap.

Still gated on `pre-run-check` (the `ready`/`verified` label or 4+ merged PRs), so nothing beyond the gate has run.
